### ISSUE TITLE
Increase Polygon precision

### ISSUE THREAD

**Issue by JosueDavalos** [ultralytics/ultralytics#15897]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

cHi! 

I'm working on a project to detect fuits using YOLOv8. I saw the model has good perfomance to detect the objects (fruits). But I realized that the polygon is not so precision. See the image. 

Is there a way to improve this? I mean, maybe, increase the number of points in the polygon to capture the object perfectly? 

![image](https://github.com/user-attachments/assets/f6d112b2-c03f-4f06-a55c-da391219d770)

**Comment by Y-T-G** [comment]
Is your annotations precise?

Otherwise, the predictions from the model won't be precise.

**Comment by JosueDavalos** [comment]
sure, train dataset is very precise. I even duplicated number of points for this porpuse and I'm still having the same result

**Comment by JosueDavalos** [comment]
take a look
![image](https://github.com/user-attachments/assets/881dce3f-9375-421a-9c18-56fcf64e1d5c)

**Comment by Y-T-G** [comment]
How many images do you have and how many epochs did you train and what was the final mAP?

**Comment by JosueDavalos** [comment]
I have around 4K images
I trained around 200 epochs in my baseline models and I using this as pretrainded for my experiments. one experiment for example was duplicated the number of points in the polygon. I trained my baseline model for 32 epochs for this experiment.  I attached the results of this 
![results](https://github.com/user-attachments/assets/c7510ac5-5e9c-4a89-b483-a7a04895675e)

**Comment by Skillnoob** [comment]
Try doing inference with `retina_masks=True`
From the [docs](https://docs.ultralytics.com/modes/predict/#inference-arguments):
> Uses high-resolution segmentation masks if available in the model. This can enhance mask quality for segmentation tasks, providing finer detail.

**Comment by JosueDavalos** [comment]
It's worked, thanks!

**Comment by Skillnoob** [comment]
Great to hear!
